cmd/kubeadm/app/discovery/flags.go

```diff
@@ -76,7 +76,16 @@ func ParseURL(d *kubeadm.Discovery, s string) error {
 	case "file":
 		return file.Parse(u, d)
 	case "token":
-		return token.Parse(s, d)
+		// Make sure a valid RFC 3986 URL has been passed and parsed.
+		// See https://github.com/kubernetes/kubeadm/issues/95#issuecomment-270431296 for more details.
+		if !strings.Contains(s, "@") {
```

Comment:
Does this work? I'm surprised we don't return an error on line 71 if we hit this. I guess that's because we store port as a string.